### PR TITLE
ICD Tagging

### PR DESCRIPTION
**Pull request by stevestrutt**
Implement Global Tagging on ICD resources. Add, Read, Delete and Update tags. Implements framework for adding tagging support to Resource Manager service instances.  
- FYI in vendoring the latest bluemix-go package, govendor appears to have added new records to vendor.json for each bluemix-go sub-package.

### CODE REVIEW

File: ibm/resource_ibm_database.go [hunk 1]
```diff
@@ -607,7 +608,20 @@ func resourceIBMDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 		return nil
 	}
 
-	d.Set("tags", instance.Tags)
```
Comment (hkantare) [inline ibm/resource_ibm_database.go]: Can we move get tags also to separate function like update..so we can use in multiple resources in future

File: examples/ibm-database/provider.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+variable "bluemix_api_key" {
+  description = "Enter your IBM Cloud API Key, you can get your IBM Cloud API key using: https://console.bluemix.net/iam#/apikeys"
+}
+
+provider "ibm" {
+  softlayer_username = "${var.softlayer_username}"
+  softlayer_api_key  = "${var.softlayer_api_key}"
+  bluemix_api_key    = "${var.bluemix_api_key}"
+  region             = "eu-gb"
+}
```
Comment (hkantare) [inline examples/ibm-database/provider.tf]: Can we add a READMe file also which explains few lines wt example is about

Comment (stevestrutt) [reply examples/ibm-database/provider.tf]: Added README.md

File: ibm/resource_ibm_database.go (hunk 1 above)
Comment (stevestrutt) [reply ibm/resource_ibm_database.go]: Moved to GetTags in structures.go

**Pull request merged**